### PR TITLE
Remove GMP

### PR DESCRIPTION
**Pull request by slaperche-scality**
It's mainly used by the tools, and it may not play well with the BSD licensing we want for the library itself.

### CODE REVIEW

File: test/fft_utest.cpp [hunk 1]
```diff
@@ -103,130 +103,6 @@ class FFTUtest {
         return vec;
     }
 
-    /**
```
Comment (slaperche-scality) [inline test/fft_utest.cpp]: I think that this makes https://github.com/scality/nttec/issues/68 outdated, if so we could close it.

File: test/fft_utest.cpp [hunk 2]
```diff
@@ -103,130 +103,6 @@ class FFTUtest {
...
-
-        // carry propagation
-        mpz_class z = 0;
-        mpz_class u;
-        for (int i = 0; i <= fft.get_n() - 1; i++) {
-            mpz_class t, b;
-            b = 10;
-            mpz_pow_ui(t.get_mpz_t(), b.get_mpz_t(), i);
-            u = mpz_class(std::to_string(sfXY->get(i)));
-            z += u * t;
```
Comment (vrancurel) [inline test/fft_utest.cpp]: Can you adapt schonage-strassen in a nttec:utils class/function (we will need it for interpolation probably) and replace carry propagation with GMP by scalar based interpolation ?

Comment (slaperche-scality) [reply test/fft_utest.cpp]: Ok, I'll add another patch in this PR (and I'll update the description in https://github.com/scality/nttec/issues/68).

**Comment by vrancurel** [comment]
OK so Schonage-Strassen is less urgent ...

**Comment by lamphamsy** [comment]
@slaperche-scality : could you please rebase the branch? :)

**Comment by slaperche-scality** [comment]
@lamphamsy done :)

**Pull request merged**